go-pkg/ds/permission.go

```diff
@@ -22,6 +28,55 @@ func (u *User) addKey(blob *protods.DataBlob, newDataOwner *User) *protods.DataB
 // 1. Generate an AES-key
 // 2. Add an entry to "Secrets"
 // 3. Encrypt data with AES key
-func (u *User) createDataBlob(data []byte) *protods.DataBlob {
-	panic("not implemented")
+func (u *User) createDataBlob(data []byte) (*protods.DataBlob, error) {
+	//HiddenDataKey creation
+	dataKey := encrypt.GenAes128Key()
+
+	ephemeralPrivKey, err := ethereum.GenerateKey()
+	if err != nil {
+		return nil, errors.New("creation of ephemeral private key failed")
+	}
+	sharedSecretX, sharedSecretY := ephemeralPrivKey.PublicKey.Add(
```

Comment:
Why `add`?

It should be `privkey * ephemeralPubkey`, where the * is a scalar multiplication of a curve point.

Reply:
Ah, got it. I missed the first part that we only represent the public key as point on the curve. Thanks a lot for the detailed explanation!